x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/225_query_ruleset_list.yml

```diff
@@ -0,0 +1,161 @@
+                  value: kibana
+              actions:
+                ids:
+                  - 'id3'
+                  - 'id4'
+
+  - do:
+      query_ruleset.put:
+        ruleset_id: test-query-ruleset-2
+        body:
+          ruleset_id: test-query-ruleset-2
+          rules:
+            - rule_id: query-rule-id1
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: elastic
+              actions:
+                ids:
+                  - 'id1'
+                  - 'id2'
+            - rule_id: query-rule-id2
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: kibana
+              actions:
+                ids:
+                  - 'id3'
+                  - 'id4'
+            - rule_id: query-rule-id3
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: logstash
+              actions:
+                ids:
+                  - 'id5'
+                  - 'id6'
+
+  - do:
+      query_ruleset.put:
+        ruleset_id: test-query-ruleset-3
+        body:
+          ruleset_id: test-query-ruleset-3
+          rules:
+            - rule_id: query-rule-id1
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: elastic
+              actions:
+                ids:
+                  - 'id1'
+                  - 'id2'
+            - rule_id: query-rule-id2
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: kibana
+              actions:
+                ids:
+                  - 'id3'
+                  - 'id4'
+            - rule_id: query-rule-id3
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: logstash
+              actions:
+                ids:
+                  - 'id5'
+                  - 'id6'
+            - rule_id: query-rule-id4
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: beats
+              actions:
+                ids:
+                  - 'id7'
+                  - 'id8'
+
+---
+"List Query Rulesets":
+  - do:
+      query_ruleset.list: { }
+
+  - match: { count: 3 }
+
+  # Alphabetical order by ruleset_id for results
+  - match: { results.0.ruleset_id: "test-query-ruleset-1" }
+  - match: { results.0.num_rules: 2 }
+
+  - match: { results.1.ruleset_id: "test-query-ruleset-2" }
+  - match: { results.1.num_rules: 3 }
+
+  - match: { results.2.ruleset_id: "test-query-ruleset-3" }
+  - match: { results.2.num_rules: 4 }
```

Comment:
Naming Nit: Maybe `rules_count` would be better instead of `num_rules`, as we already use `count` as the overall count. Not blocking in any way.

Reply:
I like that, thanks!